hazelcast-jet-core/src/main/java/com/hazelcast/jet/aggregate/AllOfAggregationBuilder.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.aggregate;
+
+import com.hazelcast.jet.datamodel.ItemsByTag;
+import com.hazelcast.jet.datamodel.Tag;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.hazelcast.jet.datamodel.Tag.tag;
+
+/**
+ * TODO
+ *
+ * Use {@link AggregateOperations#allOfBuilder()} to create.
+ *
+ * @param <T>
+ */
+public final class AllOfAggregationBuilder<T> {
+
+    private List<Tag> tags = new ArrayList<>();
+    private List<AggregateOperation1> operations = new ArrayList<>();
+
+    AllOfAggregationBuilder() { }
+
+    public <R> Tag<R> add(AggregateOperation1<? super T, ?, R> operation) {
+        operations.add(operation);
+        Tag<R> tag = tag(tags.size());
+        tags.add(tag);
+        return tag;
+    }
+
+    public AggregateOperation1<T, Object[], ItemsByTag> build() {
```

Comment:
we could also add an optional finisher function here, something like `Function<ItemsByTag, R>`